plugins/search/providers/elastic_search/ElasticSearchPlugin.php

```diff
@@ -185,4 +186,40 @@ public static function getEnums($baseEnumName = null)
 		
 		return array();
 	}
+
+	public static function canExecuteFilter(KalturaRelatedFilter $filter, KalturaDetachedResponseProfile $responseProfile = null)
+	{
+		return !$responseProfile && self::isValidClientsTagsForFilterExecutor() && ESearchEntryQueryFromFilter::canTransformFilter($filter->toObject());
+	}
+
+	public static function executeFilter(KalturaRelatedFilter $filter, KalturaFilterPager $pager)
+	{
+		$coreFilter = $filter->toObject();
+		$corePager = $pager->toObject();
+		$esearchAdapter = new ESearchEntryQueryFromFilter();
```

Comment:
Get the elastic filter adpater type from filter factory baserd on filter instance